main/src/ca/uwaterloo/flix/language/phase/unification/SimpleBoolFormulaAlgClassic.scala

```diff
@@ -0,0 +1,230 @@
+/*
```

Comment:
What's the plan for this copy paste?

Reply:
My idea is this:

- Remove `toType` and `fromType` from `BoolAlg`, instead replacing it with something like `convert`, which converts it to another `BoolAlg` representation.
- Make a `BoolAlg` implementation for `Type`/`Bool` and `Type`/`Eff`
- Use these conversions instead in what will be `EffUnification` and `BoolUnification`

We maintain the shortcuts in `EffUnification`